spec/tbd_tests_spec.rb

```diff
@@ -989,6 +990,53 @@
     model.save(file, true)
   end
 
+  it "can invalidate KIVA inputs (smalloffice)" do
+    TBD.clean!
+    argh = {}
+    translator = OpenStudio::OSVersion::VersionTranslator.new
+    file = File.join(__dir__, "files/osms/in/smalloffice.osm")
+    path = OpenStudio::Path.new(file)
+    model = translator.loadModel(path)
+    expect(model.empty?).to be(false)
+    model = model.get
+
+    # Reset all ground-facing floor surfaces as "foundations".
+    model.getSurfaces.each do |s|
+      next unless s.outsideBoundaryCondition.downcase == "ground"
+      expect(s.construction.empty?).to be(false)
+      construction = s.construction.get
+      expect(s.setOutsideBoundaryCondition("Foundation")).to be(true)
+      expect(s.setConstruction(construction)).to be(true)
+    end
+
+    argh[:option  ] = "poor (BETBG)"
+    argh[:gen_kiva] = true
+    json = TBD.process(model, argh)
+    expect(json.is_a?(Hash)).to be(true)
+    expect(json.key?(:io)).to be(true)
+    expect(json.key?(:surfaces)).to be(true)
+    io       = json[:io      ]
+    surfaces = json[:surfaces]
+    expect(TBD.status).to eq(ERR)
+    expect(TBD.logs.size).to eq(5)
+
+    TBD.logs.each do |log|
+      expect(log[:message].include?("KIVA requires standard mat")).to be(true)
+    end
+
+    expect(io.nil?).to be(false)
+    expect(io.is_a?(Hash)).to be(true)
+    expect(io.empty?).to be(false)
+    expect(surfaces.nil?).to be(false)
+    expect(surfaces.is_a?(Hash)).to be(true)
+    expect(surfaces.size).to eq(43)
+
+    surfaces.values.each { |s| expect(s.key?(:kiva)).to be(false) }
```

Comment:
While invalid construction/material errors are indeed caught and KIVA is consequently disabled, TBD will nonetheless process thermal bridging. It is preferable/prudent to (in sequence):

- have TBD generate KIVA inputs in isolation ... i.e. with the "(non thermal bridging)" option
- have SDK users edit TBD-generated KIVA inputs ... e.g. adjust slab perimeter insulation width
- have TBD process thermal bridging for outdoor-facing opaque surfaces